### PR TITLE
Fast rollback fixes

### LINKED ISSUE #89: If a rollback is triggered in the middle of a deployment, the selector is changed to the (soon-to-be) deleted deployment.
From our tests on 7/27 (here are [the full details](https://invisionapp.quip.com/npgPANELpg94#GTMACAyuUZN)).

We ran 5 forward-deployments that worked perfectly, which left the deployed version as `97ea2c8d4011e76e732fa839ceaa3b46085be835`.

Next, we started deployment 6 (UUID: `6e371753-416f-496f-9de2-4a3fa46538ea`), which began to deploy the build from commit `a1326bab30fbf5eac7331b86b1e9eda3096fd846`. About 60 seconds after the deployment had started, we ran the rollback command.

It's unclear as to exactly the timing of the events, but the TL;DR is that the `a1326bab30fbf5eac7331b86b1e9eda3096fd846` deployment was targeted to be deleted (since we asked for it to be rolled back), but the selector was changed to this deployment, resulting in CF projects going completely offline in staging.

So we just need to see if there's a timing issue here, and to see if we can add any additional checks to prevent deployments being deleted from receiving the selector that directs traffic.

### PR DESCRIPTION
**Pull request by chesleybrown**
# What

- Fixes #89 
- We were not re-throwing the error after the availability check detected a failure and thus it would always think it was successfully
- Added LastUpdated annotation
- Improved logging messaging before the pre-deploy check

# Testing

- Verified functional tests are passing locally

### CODE REVIEW

File: src/lib/annotator/annotator.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const mustBeUnique = [
 
 class Annotator {
 	constructor(options) {
+		this.start = new Date();
```
Comment (c1freitas) [inline src/lib/annotator/annotator.js]: This Annotator gets created at the beginning of the deployment?

File: src/lib/manifests.js [hunk 2]
```diff
@@ -440,6 +440,7 @@ class Manifests extends EventEmitter {
 																				status: "FAILURE",
 																				manifest: manifest
 																			});
+																			throw err;
```
Comment (c1freitas) [inline src/lib/manifests.js]: 👍

File: src/lib/strategy/strategies/fast-rollback.js [hunk 3]
```diff
@@ -103,6 +103,21 @@ class FastRollback extends EventEmitter {
...
 				.then(() => {
+					return this.kubectl.get("service", manifestName);
+				})
+				.then((result) => {
+					// If it does not have a last update annotation yet then continue
+					if (!_.has(result, ["metadata", "annotations", Annotations.LastUpdated])) {
+						return;
+					}
+					// Check if LastUpdated is newer than when this deploy started
+					const currentServiceLastUpdated = new Date(result.metadata.annotations[Annotations.LastUpdated]).getTime();
```
Comment (c1freitas) [inline src/lib/strategy/strategies/fast-rollback.js]: Do we need a fudge factor to adjust for any time drift between pods? Probably a premature optimization :)

File: src/lib/annotator/annotator.js (hunk 1 above)
Comment (chesleybrown) [inline src/lib/annotator/annotator.js]: It's created at the beginning for each cluster it's deploying to.

File: src/lib/manifests.js (hunk 2 above)
Comment (chesleybrown) [inline src/lib/manifests.js]: Can't believe this was broken for so long 😢

File: src/lib/strategy/strategies/fast-rollback.js (hunk 3 above)
Comment (chesleybrown) [inline src/lib/strategy/strategies/fast-rollback.js]: Yeah, typically we're not doing dozens of deploys a second but rather 1 deployment and possibly one rollback during the deployment. So I do think we have to worry about it here.

**Pull request merged**